lib/ecto/schema.ex

```diff
@@ -255,6 +255,8 @@ defmodule Ecto.Schema do
   * `__schema__(:primary_key)` - Returns a list of primary key fields (empty if there is none);
 
   * `__schema__(:fields)` - Returns a list of all non-virtual field names;
+  * `__schema__(:db_fields)` - Returns a list of all field names as stored in db;
+  * `__schema__(:aliases)` - Returns a list of fields with their changed source;
```

Comment:
@michalmuskala - I agree `field_aliases` is better than `aliases`, I'll rename the method. But i am confused about the `field_alias`. Do you want me to replace the option `:source` ? 

I have intentionally kept it as `source` because we use it for defining name of the schema as well. I think we could better it with `field_source`. (You can check the discussions it the same PR for other suggestions as well)

Moreover, i think `field_alias` would also require the following change. Currently,

```
field :first_name, :string, source: :fn
```
will become:
```
field :fn, :string, field_alias: :first_name
```

Since this is something big, I wanted to know your thoughts on this.